docs/tests.md

```diff
@@ -0,0 +1,439 @@
+Then MHC creating, such MHC is configured to observe this condition.
+Test waits for a `RemediationRestricted` event on the machineset and checks that no machines were deleted.
+
+This test exercises MHC behaviour that prevents remediation if a number of machines marked as unhealthy more than a number in `maxUnhelthy` parameter.
+
+### Autoscaler
+
+#### Autoscaler should use a ClusterAutoscaler that has 100 maximum total nodes count It scales from/to zero
+* Parallel: No
+* ExecTime: >7m on AWS
+* Recommendation:
+  * amount of desired machines might be reduced from 3 to 2
+
+Within this test machineset with the initial desired 0 replicas and with a specific set of labels is created.
+After, a Job with 3 desired replicas is attempting to be scheduled on nodes with labels configured for previously created machineset.
+
+The test is expecting machineset to be scaled up to 3 machines and accept the payload. Then, created Job is removing and the machineset is expected to be scaled down back to 0 replicas. 
+
+Test considered failed if scale up or down did not happen within 15 min timeout.
+
+#### Autoscaler should use a ClusterAutoscaler that has 100 maximum total nodes count cleanup deletion information after scale down [Slow]
+* Parallel: No
+* ExecTime: ~13m on AWS
+* Recommendation:
+  * This test is a bit cryptic at first glance and requires at least some in-code explanation about the meaning of checks for taints and annotations presence. Having some documentation references will be beneficial here.
+  * amount of desired machines might be reduced from 3 to 2 for each machineset
+
+Within this test machineset with 1 desired replica and with a specific set of labels is created.
+After, a Job with 6 desired replicas is attempting to be scheduled on nodes with labels configured for previously created machinesets.
+
+The test is expecting both machinesets to be scaled up to 3 machines and accept the payload. Then, created Job is removing and the machineset is expected to be scaled down back to 1 replica.
+
+After machinesets will be scaled back to 1 replica test checks if no cluster-autoscaler specific annotations are presented on machines and no autoscaler specific taints are presented on respective nodes from these machinesets.
+
+Test considered failed if scale up or down did not happen, or after scaling down there are taints/annotations leftovers within a 3 minutes timeout.
+
+#### Autoscaler should use a ClusterAutoscaler that has 12 maximum total nodes count and balance similar nodes enabled scales up and down while respecting MaxNodesTotal [Slow][Serial]
+* Parallel: No
+* ExecTime: ~13m on AWS
+* Recommendation:
+  * Since this test does not leverage or checking `balance similar nodes` feature, this might be extracted to separate group with reducing `MaxNodesTotal` number to ~8. This should speed up this test a bit and reduce resource consumption.
```

Comment:
Reducing the total here seems reasonable, I wonder if 8 was chosen for some specific reason before?

Reply:
12 you mean? I don't know. Looking into tests I didn't figure out any specific reason. @elmiko maybe you know?

Reply:
we could certainly reduce this number, i think i chose 12 because it was inspired by the original values we were using to grow the machine sets.

Reply:
looking at @RadekManak PR, i remember why it was 12. 6 for the original cluster + 6 for the new workload we created.